pycalphad/model.py

```diff
@@ -1358,6 +1358,78 @@ def _pure_element_test(constituent_array):
             reference_contrib = Add(*terms)
             referenced_value = getattr(self, out) - reference_contrib
             setattr(self, fmt_str.format(out), referenced_value)
+    
+    def volume_energy(self, dbe):
+        """
+        Return the volumetric contribution in symbolic form. Follows the approach by Lu, Selleby, and Sundman [1].
+
+        Parameters
+        ----------
+        dbe : Database
+            Database containing the relevant parameters.
+        
+        Notes
+        -----
+        The high-pressure portion of the model is not currently implemented.
+        The parameters needed for it are queried from the database; however, calculations
+        are reserved for future implementation.
+
+        References
+        ----------
+        [1] Lu, Selleby, and Sundman, Calphad, Vol. 29, No. 1, 2005, doi:10.1016/j.calphad.2005.04.001.
+        """
+
+        phase = dbe.phases[self.phase_name]
+        param_search = dbe.search
+
+        V0_param_query = (
+            (where('phase_name') == phase.name) & \
+            (where('parameter_type') == 'V0') & \
+            (where('constituent_array').test(self._array_validity))
+        )
+
+        VA_param_query = (
+            (where('phase_name') == phase.name) & \
+            (where('parameter_type') == 'VA') & \
+            (where('constituent_array').test(self._array_validity))
+        )
+
+        VK_param_query = (
+            (where('phase_name') == phase.name) & \
+            (where('parameter_type') == 'VK') & \
+            (where('constituent_array').test(self._array_validity))
+        )
+
+        VC_param_query = (
+            (where('phase_name') == phase.name) & \
+            (where('parameter_type') == 'VC') & \
+            (where('constituent_array').test(self._array_validity))
+        )
+
+        V0 = self.redlich_kister_sum(phase, param_search, V0_param_query)
+        VA = self.redlich_kister_sum(phase, param_search, VA_param_query)
+        VK = self.redlich_kister_sum(phase, param_search, VK_param_query)
+        VC = self.redlich_kister_sum(phase, param_search, VC_param_query)
+
+        # nonmagnetic contribution to volume
+        V_p0 = V0*exp(VA)
+
+        # magnetic contribution to volume
+        G_mag = self.models.get('mag')
+        V_mag = G_mag.diff(v.P)
+
+        self.MV = self.molar_volume = V_p0 + V_mag
+        self.volume_energy = S.Zero
```

Comment:
Change `self.volume_energy` to just be a function-local variable. As written, this overwrites the volume energy function itself, which is not intended behavior. At runtime users can directly access the volume energy contribution via `Model(...).models['vol']`.

Reply:
@richardotis, I believe my last commit addressed this issue.

Reply:
I added a comment by the test with how to fix it. You just need to change the name of `self.volume_energy` to `self.something_else` so that it doesn't conflict with the name of the function.